File: website/src/data/manpages.ts
import zagMain from "../../../zag-agent/man/zag.md?raw";
import helpAgent from "../../../zag-agent/man/help-agent.md?raw";
import man from "../../../zag-agent/man/man.md?raw";
import run from "../../../zag-agent/man/run.md?raw";
import exec from "../../../zag-agent/man/exec.md?raw";
import review from "../../../zag-agent/man/review.md?raw";
import config from "../../../zag-agent/man/config.md?raw";
import session from "../../../zag-agent/man/session.md?raw";
import listen from "../../../zag-agent/man/listen.md?raw";
import search from "../../../zag-agent/man/search.md?raw";
import input from "../../../zag-agent/man/input.md?raw";
import output from "../../../zag-agent/man/output.md?raw";
import status from "../../../zag-agent/man/status.md?raw";
import log from "../../../zag-agent/man/log.md?raw";
import events from "../../../zag-agent/man/events.md?raw";
import summary from "../../../zag-agent/man/summary.md?raw";
import orchestration from "../../../zag-agent/man/orchestration.md?raw";
import spawn from "../../../zag-agent/man/spawn.md?raw";
import wait from "../../../zag-agent/man/wait.md?raw";
import collect from "../../../zag-agent/man/collect.md?raw";
import pipe from "../../../zag-agent/man/pipe.md?raw";
import cancel from "../../../zag-agent/man/cancel.md?raw";
import retry from "../../../zag-agent/man/retry.md?raw";
import broadcast from "../../../zag-agent/man/broadcast.md?raw";
import watch from "../../../zag-agent/man/watch.md?raw";
import subscribe from "../../../zag-agent/man/subscribe.md?raw";
import ps from "../../../zag-agent/man/ps.md?raw";
import gc from "../../../zag-agent/man/gc.md?raw";
import env from "../../../zag-agent/man/env.md?raw";
import whoami from "../../../zag-agent/man/whoami.md?raw";
import skills from "../../../zag-agent/man/skills.md?raw";
import mcp from "../../../zag-agent/man/mcp.md?raw";
import capability from "../../../zag-agent/man/capability.md?raw";
import serve from "../../../zag-agent/man/serve.md?raw";
import connect from "../../../zag-agent/man/connect.md?raw";

export interface ManPage {
  slug: string;
  title: string;
  content: string;
}

export interface ManPageGroup {
  label: string;
  pages: ManPage[];
}

export const manPageGroups: ManPageGroup[] = [
  {
    label: "Overview",
    pages: [
      { slug: "zag", title: "zag", content: zagMain },
      { slug: "help-agent", title: "zag --help-agent", content: helpAgent },
      { slug: "man", title: "zag man", content: man },
    ],
  },
  {
    label: "Core Commands",
    pages: [
      { slug: "run", title: "zag run", content: run },
      { slug: "exec", title: "zag exec", content: exec },
      { slug: "review", title: "zag review", content: review },
      { slug: "config", title: "zag config", content: config },
    ],
  },
  {
    label: "Sessions",
    pages: [
      { slug: "session", title: "zag session", content: session },
      { slug: "listen", title: "zag listen", content: listen },
      { slug: "search", title: "zag search", content: search },
      { slug: "input", title: "zag input", content: input },
      { slug: "output", title: "zag output", content: output },
      { slug: "status", title: "zag status", content: status },
      { slug: "log", title: "zag log", content: log },
      { slug: "events", title: "zag events", content: events },
      { slug: "summary", title: "zag summary", content: summary },
    ],
  },
  {
    label: "Orchestration",
    pages: [
      { slug: "orchestration", title: "zag orchestration", content: orchestration },
      { slug: "spawn", title: "zag spawn", content: spawn },
      { slug: "wait", title: "zag wait", content: wait },
      { slug: "collect", title: "zag collect", content: collect },
      { slug: "pipe", title: "zag pipe", content: pipe },
      { slug: "cancel", title: "zag cancel", content: cancel },
      { slug: "retry", title: "zag retry", content: retry },
      { slug: "broadcast", title: "zag broadcast", content: broadcast },
      { slug: "watch", title: "zag watch", content: watch },
      { slug: "subscribe", title: "zag subscribe", content: subscribe },
    ],
  },
  {
    label: "Process Management",
    pages: [
      { slug: "ps", title: "zag ps", content: ps },
      { slug: "gc", title: "zag gc", content: gc },
      { slug: "env", title: "zag env", content: env },
      { slug: "whoami", title: "zag whoami", content: whoami },
    ],
  },
  {
    label: "Extensions",
    pages: [
      { slug: "skills", title: "zag skills", content: skills },
      { slug: "mcp", title: "zag mcp", content: mcp },
      { slug: "capability", title: "zag capability", content: capability },
    ],
  },
  {
    label: "Remote Access",
    pages: [
      { slug: "serve", title: "zag serve", content: serve },
      { slug: "connect", title: "zag connect", content: connect },
    ],
  },
];

export const manPages: ManPage[] = manPageGroups.flatMap((group) => group.pages);

export function getManPageBySlug(slug: string): ManPage | undefined {
  return manPages.find((page) => page.slug === slug);
}

File: zag-agent/src/manpages.rs
//! Built-in manpages and the `--help-agent` reference document.
//!
//! Markdown sources live in `zag-agent/man/` and are embedded at compile
//! time via [`include_str!`]. The CLI's `zag man <command>` and
//! `zag --help-agent` subcommands read from this module, and library
//! callers can use the same content programmatically.
//!
//! # Example
//!
//! ```no_run
//! use zag_agent::manpages;
//!
//! // Full AI-oriented reference
//! let help = manpages::HELP_AGENT;
//! println!("{help}");
//!
//! // Per-command manpage
//! if let Some(md) = manpages::manpage(Some("review")) {
//!     println!("{md}");
//! }
//!
//! // Enumerate everything that has a manpage
//! for name in manpages::manpage_names() {
//!     println!("{name}");
//! }
//! ```

/// Top-level CLI overview.
pub const ZAG: &str = include_str!("../man/zag.md");
/// AI-oriented CLI reference, printed by `zag --help-agent`.
pub const HELP_AGENT: &str = include_str!("../man/help-agent.md");
pub const MAN: &str = include_str!("../man/man.md");
pub const RUN: &str = include_str!("../man/run.md");
pub const EXEC: &str = include_str!("../man/exec.md");
pub const REVIEW: &str = include_str!("../man/review.md");
pub const PLAN: &str = include_str!("../man/plan.md");
pub const CONFIG: &str = include_str!("../man/config.md");
pub const SESSION: &str = include_str!("../man/session.md");
pub const CAPABILITY: &str = include_str!("../man/capability.md");
pub const DISCOVER: &str = include_str!("../man/discover.md");
pub const LISTEN: &str = include_str!("../man/listen.md");
pub const SKILLS: &str = include_str!("../man/skills.md");
pub const MCP: &str = include_str!("../man/mcp.md");
pub const PS: &str = include_str!("../man/ps.md");
pub const SEARCH: &str = include_str!("../man/search.md");
pub const INPUT: &str = include_str!("../man/input.md");
pub const BROADCAST: &str = include_str!("../man/broadcast.md");
pub const WHOAMI: &str = include_str!("../man/whoami.md");
pub const WAIT: &str = include_str!("../man/wait.md");
pub const SPAWN: &str = include_str!("../man/spawn.md");
pub const STATUS: &str = include_str!("../man/status.md");
pub const COLLECT: &str = include_str!("../man/collect.md");
pub const ENV: &str = include_str!("../man/env.md");
pub const PIPE: &str = include_str!("../man/pipe.md");
pub const EVENTS: &str = include_str!("../man/events.md");
pub const CANCEL: &str = include_str!("../man/cancel.md");
pub const SUMMARY: &str = include_str!("../man/summary.md");
pub const WATCH: &str = include_str!("../man/watch.md");
pub const SUBSCRIBE: &str = include_str!("../man/subscribe.md");
pub const LOG: &str = include_str!("../man/log.md");
pub const OUTPUT: &str = include_str!("../man/output.md");
pub const RETRY: &str = include_str!("../man/retry.md");
pub const GC: &str = include_str!("../man/gc.md");
pub const SERVE: &str = include_str!("../man/serve.md");
pub const CONNECT: &str = include_str!("../man/connect.md");
pub const USER: &str = include_str!("../man/user.md");
pub const USAGE: &str = include_str!("../man/usage.md");
pub const ORCHESTRATION: &str = include_str!("../man/orchestration.md");

/// Names of every manpage accepted by [`manpage`], in a stable order suited
/// for user-facing listings.
pub const MANPAGE_NAMES: &[&str] = &[
    "zag",
    "run",
    "exec",
    "review",
    "plan",
    "config",
    "session",
    "capability",
    "discover",
    "listen",
    "man",
    "skills",
    "mcp",
    "ps",
    "search",
    "input",
    "broadcast",
    "whoami",
    "wait",
    "spawn",
    "status",
    "collect",
    "env",
    "pipe",
    "events",
    "cancel",
    "summary",
    "watch",
    "subscribe",
    "log",
    "output",
    "retry",
    "gc",
    "serve",
    "connect",
    "disconnect",
    "user",
    "usage",
    "orchestration",
];

/// Return the manpage content for the given command name, or `None` if the
/// name is not recognised. `None` / empty / `"zag"` all return the top-level
/// [`ZAG`] manpage. `"disconnect"` aliases to [`CONNECT`].
pub fn manpage(command: Option<&str>) -> Option<&'static str> {
    Some(match command.unwrap_or("zag") {
        "" | "zag" => ZAG,
        "run" => RUN,
        "exec" => EXEC,
        "review" => REVIEW,
        "plan" => PLAN,
        "config" => CONFIG,
        "session" => SESSION,
        "capability" => CAPABILITY,
        "discover" => DISCOVER,
        "listen" => LISTEN,
        "man" => MAN,
        "skills" => SKILLS,
        "mcp" => MCP,
        "ps" => PS,
        "search" => SEARCH,
        "input" => INPUT,
        "broadcast" => BROADCAST,
        "whoami" => WHOAMI,
        "wait" => WAIT,
        "spawn" => SPAWN,
        "status" => STATUS,
        "collect" => COLLECT,
        "env" => ENV,
        "pipe" => PIPE,
        "events" => EVENTS,
        "cancel" => CANCEL,
        "summary" => SUMMARY,
        "watch" => WATCH,
        "subscribe" => SUBSCRIBE,
        "log" => LOG,
        "output" => OUTPUT,
        "retry" => RETRY,
        "gc" => GC,
        "serve" => SERVE,
        "connect" | "disconnect" => CONNECT,
        "user" => USER,
        "usage" => USAGE,
        "orchestration" => ORCHESTRATION,
        _ => return None,
    })
}

/// Names of every command for which [`manpage`] returns `Some`.
pub fn manpage_names() -> &'static [&'static str] {
    MANPAGE_NAMES
}

#[cfg(test)]
#[path = "manpages_tests.rs"]
mod tests;
